core/dmux.go

```diff
@@ -311,7 +378,172 @@ func simpleSetup(size, qsize int, sink Sink) ([]chan interface{}, *sync.WaitGrou
 		go func(index int) {
 			sk := sink.Clone()
 			for msg := range ch[index] {
-				sk.Consume(msg)
+				sk.Consume(msg, math.MaxInt32)
+			}
+			wg.Done()
+		}(i)
+	}
+	return ch, wg
+}
+
+func simpleSetupWithSideline(size, qsize int, sink Sink, sideline Sideline, sidelineImpl interface{}) ([]chan interface{}, *sync.WaitGroup) {
```

Comment:
Change `sidelineImpl` to  proper type here itself, to avoid casting it later.

Reply:
will change this.